lib/layout/force-directed-layout.js

```diff
@@ -0,0 +1,202 @@
+/**
+ * force-directed-layout.js
+ *
+ * This module contains logic for performing the force-directed layout
+ * algorithm on a graph.
+ *
+ * @author John O'Brien.
+ */
+ 
+var CardinalPortPosition = require('./../model/cardinalPortPosition');
+var change = false;
+//ideal length
+var iLength = 500;
+//force strength
+var forceStr = 1.0/3.0;
```

Comment:
JS has only one number type, which means that `1/3` is real division, not integer division. There's no need for the `.0`.